### PR TITLE
[v6.2] SSH Articles Refactor

### PR DESCRIPTION
**Pull request by inertial-frame**
This extracts out SSH-related content used during the 6.2 Server Access Section, improves it, and combines content from the User and Admin Guides.

- [x] Went through and validated the SSH - validated.
- [x] Further validating OpenSSH client article - validated.
- [x] Need to further validate the OpenSSH recording article - validated.
- [x] In addressing the above, I'll try to better address: https://github.com/gravitational/teleport/issues/3734 from a documentation standpoint - we can improve both the CLI option and the documentation steps.
- [x] Improve Windows SSH client documentation: Putty CAC (supports CA), WLS 1/2 supports OpenSSH on Windows 10, Cygwin on Windows, etc. This could use some extra care since it's been an ongoing point of confusion!
- [ ] May get to: https://github.com/gravitational/teleport/issues/6958 in this PR or a follow-up - this is partly done - some of the port hardening items I did not explicitly attend to. Some of the 7.0 features will likely require updating the include.
- [x] Encourage backing up an instance (with configuration prior to Teleport setup). `Admonition -> Warning`.
- [ ] We may want to update the docs to include sharing as well: https://github.com/gravitational/teleport/issues/1968 - the process to do this has been vastly simplified. I believe I was able to connect using two different test users logging in through `tsh`, record both of their interactions, then replay them.

### CODE REVIEW

File: docs/pages/server-access/guides/ssh.mdx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+     caption="Both Nodes in the Web UI"
+   >
+     ![Both Nodes in the Web UI](../../../img/server-access/teleport_ui.png)
+   </Figure>
+
+## Connect to a cluster using tsh ssh
+
+Teleport SSH is robust enough that you can do all the things you'd do with OpenSSH, without it.
+
+To connect to a cluster (`auth.example.com`) with public proxy `auth.example.com:443` as user `auth-admin` using `tsh ssh`:
```
Comment (benarent) [inline docs/pages/server-access/guides/ssh.mdx:69]: When do we create `auth-admin`?  and is `access` and `editor` enough permissions for that user?

Comment (inertial-frame) [reply docs/pages/server-access/guides/ssh.mdx]: Thanks that's a great point - I've rewritten that part to include more recent items from the Getting Started document. This is mostly temporary staging for moving a lot of content around. Links break, images are shifting, etc.

File: docs/pages/server-access/guides/openssh.mdx [hunk 2]
```diff
@@ -69,6 +69,100 @@ Teleport OpenSSH supports:
 - IPv4 `ubuntu@184.45.45.30`
 - IPv6 `root@2001:db8::2`
 
+## Using OpenSSH client
```
Comment (inertial-frame) [inline docs/pages/server-access/guides/openssh.mdx:72]: This section is moved from the User Manual but is mostly redundant here. I will prune this quite a bit.

**Comment by klizhentas** [comment]
@inertial-frame for PRs that are work in progress, can you set those to `draft PR` mode?

File: docs/pages/server-access/guides/openssh-recording.mdx [hunk 3]
```diff
@@ -143,26 +143,27 @@ ssh-keygen -L -f api.example.com-cert.pub
...
+      Critical Options: (none)
+      Extensions: 
+            x-teleport-authority UNKNOWN OPTION (len 27)
+            x-teleport-role UNKNOWN OPTION (len 8)
 ```
 
 Then add the following lines to `/etc/ssh/sshd_config` on all OpenSSH nodes, and restart `sshd`.
 
-```yaml
+```bash
```
Comment (inertial-frame) [inline docs/pages/server-access/guides/openssh-recording.mdx:166]: These are technically lines that are added to the `sshd_config` but I think we prefer to render this with `bash` per a comment I saw.

File: docs/pages/server-access/includes/ssh-hardening.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+## OpenSSH hardening
+
+This section describes general ways to secure your OpenSSH client and server connections.
```
Comment (inertial-frame) [inline docs/pages/server-access/includes/ssh-hardening.mdx:3]: This addresses some of the #6958 items. I can add to this now, add to this once more 7.0 features are released, etc.

File: docs/pages/user-manual.mdx [hunk 5]
```diff
@@ -458,10 +419,10 @@ tsh join <session_ID>
 ## Connecting to SSH clusters behind firewalls
```
Comment (inertial-frame) [inline docs/pages/user-manual.mdx:419]: I think this should go into the security architecture article along with several Bastion, Jumphost, and firewall-related items.

File: docs/pages/server-access/guides/ssh.mdx [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+   ```bash
+   # Log in through tsh
+   tsh login --proxy=tele.example.com:443 --user=tele-admin
+   ```
+   
+   You'll be prompted to supply the password and One Time Passcode you set up previously for `tele-admin`.
+
+2. `tele-admin` will now see:
+
+   ```bash
```
Comment (inertial-frame) [inline docs/pages/server-access/guides/ssh.mdx:74]: This is an example, not a necessary role assignment.

In the future, I think we should have a separate "example-only" code-block: https://github.com/gravitational/next/issues/275 clearly indicating that it's an example and can't be copy-pasted.

File: docs/pages/server-access/guides/ssh.mdx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+  caption="Teleport SSH agent forwarding."
+>
+  ![Example cluster arrangement](../../../img/server-access/teleport-ssh.svg)
+</Figure>
+
+## Choose your SSH solution
+
+SSH tools are now commonly distributed as clients, servers, or some combination of both.  
+
+Teleport is unopinionated and agnostic about which client and/or server you choose. Admins can select specific Teleport features (like [Session Recording](../../features/enhanced-session-recording.mdx) while opting to continue using some or all of their existing SSH solutions.
```
Comment (inertial-frame) [inline docs/pages/server-access/guides/ssh.mdx:24]: Will remove extra `(`

File: docs/pages/faq.mdx [hunk 8]
```diff
@@ -22,10 +22,9 @@ available unless the Teleport SSH daemon is present on all cluster nodes.
 
 ### Can I use OpenSSH with a Teleport cluster?
 
-Yes, this question comes up often and is related to the previous one. Take a
-look at [Using OpenSSH client](user-manual.mdx#using-openssh-client)
-section in the User Manual and [Using OpenSSH servers](admin-guide.mdx) in the
-Admin Manual.
+Yes, this question comes up often and is related to the previous one. 
+
+Take a look at [Using OpenSSH client](./server-access/guides/openssh-client.mdx) and [Using OpenSSH servers](admin-guide.mdx) articles.
```
Comment (inertial-frame) [inline docs/pages/faq.mdx:27]: This should point to the SSH server article.

**Pull request closed without merging**